customer-terminate-service/domain/src/test/java/se/inera/intyg/cts/domain/service/CollectExportContentTest.java

```diff
@@ -0,0 +1,203 @@
+package se.inera.intyg.cts.domain.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static se.inera.intyg.cts.domain.util.TerminationTestDataFactory.defaultTermination;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import se.inera.intyg.cts.domain.model.Certificate;
+import se.inera.intyg.cts.domain.model.CertificateBatch;
+import se.inera.intyg.cts.domain.model.CertificateId;
+import se.inera.intyg.cts.domain.model.CertificateSummary;
+import se.inera.intyg.cts.domain.model.CertificateText;
+import se.inera.intyg.cts.domain.model.CertificateType;
+import se.inera.intyg.cts.domain.model.CertificateTypeVersion;
+import se.inera.intyg.cts.domain.model.CertificateXML;
+import se.inera.intyg.cts.domain.model.Termination;
+import se.inera.intyg.cts.domain.model.TerminationId;
+import se.inera.intyg.cts.domain.model.TerminationStatus;
+import se.inera.intyg.cts.domain.repository.InMemoryCertificateBatchRepository;
+import se.inera.intyg.cts.domain.repository.InMemoryCertificateRepository;
+import se.inera.intyg.cts.domain.repository.InMemoryCertificateTextRepository;
+import se.inera.intyg.cts.domain.repository.InMemoryTerminationRepository;
+
+class CollectExportContentTest {
+
+  private CollectExportContent collectExportContent;
+  private InMemoryTerminationRepository inMemoryTerminationRepository;
+  private InMemoryCertificateBatchRepository inMemoryCertificateBatchRepository;
+  private InMemoryCertificateRepository inMemoryCertificateRepository;
+  private InMemoryCertificateTextRepository inMemoryCertificateTextRepository;
+
+  @BeforeEach
+  void setUp() {
+    inMemoryTerminationRepository = new InMemoryTerminationRepository();
+    inMemoryCertificateBatchRepository = new InMemoryCertificateBatchRepository();
+    inMemoryCertificateRepository = new InMemoryCertificateRepository();
+    inMemoryCertificateTextRepository = new InMemoryCertificateTextRepository();
+    collectExportContent = new CollectExportContent(inMemoryTerminationRepository,
+        inMemoryCertificateBatchRepository, inMemoryCertificateRepository,
+        inMemoryCertificateTextRepository);
+  }
+
+  @Nested
+  class CollectCertificates {
```

Comment:
The only difference from mocks and InMemory-classes is that instead of mocking for example the  certificateBatchRepository.nextBatch(termination) call, I prepare the inMemory-class implementing the CertificateBatchRepository-interface with the data needed for the test.

Reply:
There is also logic in the InMemory-classes that are implementing the different interfaces that is not directly visible in the tests and may affect what is expected, These tests are actually testing CollectExportContent and all 4 InMemory-classes. The assertions verify that the CollectExportContent and the InMemory-classes work together. Kind of a risky :)